src/Hosting/TestHost/src/UpgradeFeature.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Microsoft.AspNetCore.TestHost
+{
+    internal class UpgradeFeature : IHttpUpgradeFeature
+    {
+        public bool IsUpgradableRequest => true;
```

Comment:
The presence of the feature should be enough, upgrades do not need to be possible.
```suggestion
        public bool IsUpgradableRequest => false;
```